daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Result.scala

```diff
@@ -105,8 +105,7 @@ object Result {
         ResultNeedPackage(packageId, {
           case None => ResultError(Error(s"Couldn't find package $packageId"))
           case Some(pkg) =>
-            compiledPackages.addPackage(packageId, pkg)
-            resume(pkg)
+            compiledPackages.addPackage(packageId, pkg).flatMap(_ => resume(pkg))
```

Comment:
makes sense, cc @dajmaki

Reply:
`"-P:wartremover:traverser:org.wartremover.warts.NonUnitStatements"` is must have